test/e2e/label_filter_test.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestLabelFilter(t *testing.T) {
```

Comment:
for this functionality having an e2e test is quite heavy. i would recommend adding a unit test against ResourceFilter class directly.

if this is the only e2e test that touches filtering i would modify it a bit to be more generic, eg
- test 2 filter flags in combination
- test --filter

also we do not need to actually deploy things, just have to check against diff summary since that's what filter applies to.